.github/workflows/gha-ci.yml

```diff
@@ -0,0 +1,103 @@
+name: GitHubActions CI tests
+
+on:
+  push:
+    branches: [master, dev, ci]
+  pull_request:
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        os: [ubuntu-20.04, macos-10.15]
+        build_suite: [autoconf]
+        with_coveralls: [no]
+        include:
+          # Do one build with cmake instead of autoconf
+          - os: ubuntu-20.04
+            build_suite: cmake
+            with_coveralls: no
+          # And run one build with gcov and coveralls
+          # (Does currently not work, but neither did it in travis)
+          #- os: ubuntu-20.04
+          #  build_suite: autoconf
+          #  with_coveralls: yes
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - name: checkout code
+        uses: actions/checkout@v2
+
+      - name: install deps
+        run: |
+          if (echo "${{ matrix.os }}" | grep -qE '^macos-') ; then
+            brew update
+            # freetype, gettext, icuc4 and autoconf are preinstalled
+            # and `brew install` fails if a non-uptodate version is already installed
+            #brew upgrade autoconf freetype gettext icu4c
+            brew install automake boost ffmpeg ffms2 fftw fribidi libass \
+                         lua luarocks m4 wxmac hunspell uchardet
+          else
+            #sudo apt-get update && sudo apt-get upgrade
+            sudo apt-get install -y ${{ matrix.build_suite }} \
+              autopoint build-essential g++ gcc gettext libasound2-dev \
+              libass-dev  libffms2-dev libfftw3-dev libfontconfig1-dev   \
+              libfribidi-dev libgtest-dev libhunspell-dev libicu-dev libopenal-dev       \
+              libpulse-dev libuchardet-dev libwxgtk3.0-gtk3-dev luarocks portaudio19-dev \
+              python3-pip python3-setuptools \
+              pkg-config libgl1-mesa-dev libgl-dev libfreetype6-dev libfontconfig-dev \
+              intltool libboost-all-dev
+            if [ "x${{ matrix.with_coveralls }}" = "xyes" ] ; then
+              sudo pip3 install -U cpp-coveralls
+            fi
+          fi
+
+          sudo luarocks install busted     > /dev/null
+          sudo luarocks install moonscript > /dev/null
+          sudo luarocks install uuid       > /dev/null
+
+          git submodule --quiet init
+          git submodule --quiet update vendor/googletest
+
+      - name: configure
+        run: |
+          if (echo "${{ matrix.os }}" | grep -qE '^macos-') ; then
+            export PATH="/usr/local/opt/gettext/bin:/usr/local/opt/icu4c/sbin:/usr/local/opt/icu4c/bin:$PATH";
+            export CPPFLAGS="-I/usr/local/opt/gettext/include -I/usr/local/opt/icu4c/include";
+            export LDFLAGS="-L/usr/local/opt/gettext/lib -L/usr/local/opt/icu4c/lib";
+            export ACLOCAL_PATH="/usr/local/opt/gettext/share/aclocal";
+            export PKG_CONFIG_PATH="/usr/local/opt/icu4c/lib/pkgconfig";
```

Comment:
**MacOS:**

I suspect this part is what needs tweaking to get the MacOS build working. I've poked a bit at it *(without success obviously)*:

- With the env as is (taken from Travis), it fails during linking, see: https://github.com/TheOneric/Aegisub/runs/1559097260?check_suite_focus=true#step:5:263
- If nothing is set, it fails to find a working boost+icu during configuration
- The only parts neccessary to get boost+icu working *(after which it fails with the exact same error as when everything is set is)*:
  ```
  export CPPFLAGS="-I/usr/local/opt/icu4c/include";
  export PKG_CONFIG_PATH="/usr/local/opt/icu4c/lib/pkgconfig";
  ```

Reply:
Some packages from homebrew are installed as [keg-only](https://docs.brew.sh/FAQ#what-does-keg-only-mean). They are installed without overriding the macOS-provided ones. When installing keg-only package, homebrew will print instructions on how to use these packages by setting the environment vars. These lines are copied directly from homebrew's instructions. So some of them might be unnecessary.

Reply:
The linking failure seems related to LuaJIT/LuaJIT#449

Reply:
Thanks for the hints. Unlike a proper autotools build, Aegisub's autoconf+custom Makefile combination does not keep `CPPFLAGS` set at configure-time saved for subsequent builds and env vars exported in a GHA step are normally reset after each step.

Env setup is now separated from the other steps using GHA-specific behaviour to alter the env of the following steps.